connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -348,6 +350,33 @@ public void testWakeupInCommitSyncCausesRetry() throws Exception {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testTimestampPropagation() throws Exception {
+        expectInitializeTask();
+        expectPollInitialAssignment();
+
+        expectConsumerPoll(1);
+        expectConvertMessages(1);
+
+        Capture<Collection<SinkRecord>> records = EasyMock.newCapture(CaptureType.ALL);
+
+        sinkTask.put(EasyMock.capture(records));
+
+        PowerMock.replayAll();
+
+        workerTask.initialize(TASK_CONFIG);
+        workerTask.initializeAndStart();
+        workerTask.poll(Long.MAX_VALUE);
+        workerTask.poll(Long.MAX_VALUE);
```

Comment:
@shikhar This is needed because these tests reuse a common `expectPollInitialAssignment` which *only* handles getting the member into the group and returns zero records. The subsequent expected poll call will actually retrieve the message and cause the records to be put into the sink task.